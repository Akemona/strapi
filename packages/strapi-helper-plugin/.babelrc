{
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": false,
        "loose": true,
        "targets": {
          "browsers": ["Since 2017"]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": ["@babel/plugin-proposal-class-properties"]
}
