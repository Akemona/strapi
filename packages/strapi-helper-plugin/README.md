# Strapi Helper Plugin

---

## Deprecation Warning :warning:

Hello! We have some news to share,

We’ve decided it’ll soon be time to end the support for `strapi-helper-plugin`.

After years of iterations, Strapi is going to V4 and we won’t maintain V3 packages when it’ll reach its end-of-support milestone (~end of Q3 2022).

If you’ve been using `strapi-helper-plugin` and have migrated to V4 (or if you want to), you can find the equivalent and updated version of this package at this [URL](https://github.com/strapi/strapi/tree/master/packages/core/helper-plugin) and with the following name on NPM: `@strapi/helper-plugin`.

If you’ve contributed to the development of this package, thank you again for that! We hope to see you on the V4 soon.

The Akemona team

---

## Description

Helper to develop Strapi plugins.
