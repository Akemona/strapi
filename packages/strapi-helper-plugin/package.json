{
  "name": "@akemona-org/strapi-helper-plugin",
  "publishConfig": {
    "access": "public"
  },
  "version": "3.12.4",
  "description": "Helper for Strapi plugins development",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/akemona/strapi.git",
    "directory": "packages/strapi-helper-plugin"
  },
  "engines": {
    "node": ">=14.19.1 <=18.x.x",
    "npm": ">=6.0.0"
  },
  "author": {
    "email": "strapi@akemona.com",
    "name": "Akemona team",
    "url": "https://strapi.akemona.com"
  },
  "main": "dist/strapi-helper-plugin.cjs.min.js",
  "module": "dist/strapi-helper-plugin.esm.min.js",
  "maintainers": [
    {
      "name": "Akemona team",
      "email": "strapi@akemona.com",
      "url": "https://strapi.akemona.com"
    }
  ],
  "license": "SEE LICENSE IN LICENSE",
  "scripts": {
    "prepublishOnly": "npm run build",
    "prebuild": "rimraf dist",
    "build": "rollup -c",
    "watch": "rollup -c -w",
    "test": "echo \"no tests yet\""
  },
  "devDependencies": {
    "@babel/core": "^7.14.0",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/preset-env": "^7.13.15",
    "@babel/preset-react": "^7.13.13",
    "enzyme": "^3.8.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "rimraf": "4.1.0",
    "rollup": "^1.10.0",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-filesize": "^6.0.1",
    "rollup-plugin-node-resolve": "^4.2.3",
    "rollup-plugin-postcss": "^3.1.8",
    "rollup-plugin-replace": "^2.2.0",
    "rollup-plugin-svg": "^1.0.1",
    "rollup-plugin-terser": "^4.0.4"
  },
  "dependencies": {
    "@buffetjs/core": "3.3.8",
    "@buffetjs/custom": "3.3.8",
    "@buffetjs/hooks": "3.3.8",
    "@buffetjs/icons": "3.3.8",
    "@buffetjs/styles": "3.3.8",
    "@buffetjs/utils": "3.3.8",
    "@fortawesome/fontawesome-free": "^5.15.2",
    "@fortawesome/fontawesome-svg-core": "^1.2.35",
    "@fortawesome/free-brands-svg-icons": "^5.15.2",
    "@fortawesome/free-solid-svg-icons": "^5.15.3",
    "@fortawesome/react-fontawesome": "^0.1.14",
    "bootstrap": "^4.6.0",
    "classnames": "^2.3.1",
    "immutable": "^3.8.2",
    "invariant": "^2.2.1",
    "lodash": "4.17.21",
    "moment": "^2.29.4",
    "react": "^16.14.0",
    "react-dom": "^16.9.0",
    "react-helmet": "^6.1.0",
    "react-intl": "4.5.0",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.0.0",
    "reactstrap": "8.4.1",
    "styled-components": "^5.2.3",
    "whatwg-fetch": "^3.6.2"
  },
  "gitHead": "57095241bb8bc011c97ecbc8c7274b674b01cd0f"
}
