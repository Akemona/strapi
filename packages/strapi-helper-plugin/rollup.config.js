import babel from 'rollup-plugin-babel';
import resolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import svg from 'rollup-plugin-svg';
import postcss from 'rollup-plugin-postcss';
import { terser } from 'rollup-plugin-terser';
import replace from 'rollup-plugin-replace';
import filesize from 'rollup-plugin-filesize';
import pkg from './package.json';

export default {
  input: './lib/src/index.js',
  output: [
    {
      exports: 'named',
      file: pkg.main,
      format: 'cjs',
      sourceMap: false,
      name: 'strapi-helper-plugin',
      compact: true,
      globals: {
        react: 'React',
        'react-dom': 'ReactDOM',
      },
    },
    {
      exports: 'named',
      sourceMap: false,
      file: pkg.module,
      format: 'es',
      name: 'strapi-helper-plugin',
      compact: true,
      globals: {
        react: 'React',
        'react-dom': 'ReactDOM',
      },
    },
  ],
  plugins: [
    postcss({
      modules: true,
      minimize: true,
    }),
    resolve(),
    babel({
      exclude: 'node_modules/**',
    }),
    commonjs(),
    svg(),
    terser(),
    filesize(),
    replace({
      'process.env.NODE_ENV': JSON.stringify('production'),
    }),
  ],

  external: [
    ...Object.keys(pkg.dependencies || {}),
    ...Object.keys(pkg.peerDependencies || {}),
  ],
};
