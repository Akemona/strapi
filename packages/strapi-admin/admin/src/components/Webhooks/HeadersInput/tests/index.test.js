import React from 'react';
import { shallow } from 'enzyme';
import CreatableSelect from 'react-select/creatable';
import { render, cleanup } from '@testing-library/react';
import { InputText } from '@buffetjs/core';
import { CircleButton } from 'strapi-helper-plugin';
import { IntlProvider } from 'react-intl';

import HeadersInput from '../index';

describe('Admin | components | HeadersInput', () => {
  const props = {
    name: 'headers',
    value: [
      {
        key: '',
        value: '',
      },
    ],
    onChange: jest.fn(),
    onClick: jest.fn(),
    onRemove: jest.fn(),
  };

  describe('should render properly', () => {
    afterEach(cleanup);

    it('should not crash', () => {
      shallow(<HeadersInput {...props} />);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <IntlProvider locale="en" textComponent="span">
          <HeadersInput {...props} />
        </IntlProvider>
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render as many rows as value length', () => {
      const renderedComponent = shallow(<HeadersInput {...props} />);

      expect(renderedComponent.find(CreatableSelect)).toHaveLength(1);
    });

    it('should have default onRemove', () => {
      expect(HeadersInput.defaultProps.onRemove).toBeDefined();
    });
  });

  describe('Actions', () => {
    it('should call onRemove props on remove button click', () => {
      const headers = [
        {
          key: 'Accept',
          value: 'text/html',
        },
        {
          key: 'Authorization',
          value: 'Basic YWxhZGRpbjpvcGVuc2VzYW1l',
        },
      ];
      const renderedComponent = shallow(<HeadersInput {...props} value={headers} />);

      const removeButton = renderedComponent.find(CircleButton).at(0);
      removeButton.simulate('click');

      expect(props.onRemove).toHaveBeenCalledWith(0);
    });

    it('should call onClick props on add button click', () => {
      const renderedComponent = shallow(<HeadersInput {...props} />);

      const addButton = renderedComponent.find('ul + button');
      addButton.simulate('click');

      expect(props.onClick).toHaveBeenCalledWith('headers');
    });

    it('should call the onChange props on input text change', () => {
      const renderedComponent = shallow(<HeadersInput {...props} />);

      const input = renderedComponent.find(InputText).at(0);
      const event = {
        target: { name: 'headers.0.key', value: 'test' },
      };

      input.simulate('change', event);

      expect(props.onChange).toHaveBeenCalledWith(event);
    });
  });
});
