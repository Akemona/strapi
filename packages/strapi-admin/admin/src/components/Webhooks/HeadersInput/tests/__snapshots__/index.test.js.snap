// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin | components | HeadersInput should render properly should match the snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: 100%;
  height: 3.4rem;
  padding: 0 1rem;
  font-weight: 400;
  font-size: 1.3rem;
  cursor: text;
  outline: 0;
  border: 1px solid #E3E9F3;
  border-radius: 2px;
  color: #333740;
  background-color: transparent;
}

.c2::-webkit-input-placeholder {
  color: #919BAE;
}

.c2:focus {
  border-color: #78caff;
}

.c2:disabled {
  background-color: #FAFAFB;
  cursor: not-allowed;
  color: #9ea7b8;
}

.c1 {
  position: relative;
  background-color: #ffffff;
  border-radius: 2px;
}

.c1 button[type='button'] {
  width: 3.4rem;
  height: 3.4rem;
  position: absolute;
  top: 0;
  right: 0;
  z-index: 2;
  cursor: pointer;
  outline: 0;
  border: 0;
  background: transparent;
}

.c1 button[type='button']:hover span::before {
  color: #000000;
}

.c1 button[type='button'].shown span::before {
  color: #000000;
}

.c1 button[type='button'].shown + input[type='text'] {
  padding-right: calc(3.4rem + 1rem);
}

.c1 span {
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
}

.c1 input {
  position: relative;
  z-index: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 20px;
  width: 20px;
  margin: 8px 5px 0px 0px;
  border-radius: 50%;
  border: 1px solid #e3e9f3;
  cursor: pointer;
}

.c3:focus {
  outline: 0;
}

.c3:after {
  content: '\\f068';
  font-family: FontAwesome;
  font-size: 9px;
  line-height: 18px;
  color: #007eff;
}

.c0 {
  margin-top: 12px;
  padding: 23px 24px 26px 24px;
  background-color: #fafafb;
  border-radius: 2px;
}

.c0 > ul {
  margin-bottom: 0;
  padding: 0;
  list-style-type: none;
}

.c0 > ul + button {
  padding: 0;
  color: #007eff;
  font-size: 13px;
  font-weight: 500;
  outline: 0;
}

.c0 > ul + button svg,
.c0 > ul + button span {
  vertical-align: middle;
}

.c0 > ul + button svg {
  margin-right: 10px;
}

.c0 > ul li {
  position: relative;
  padding-right: 30px;
}

.c0 > ul li:not(:first-of-type) {
  margin-bottom: 20px;
}

.c0 > ul li:last-of-type {
  margin-bottom: 6px;
}

.c0 > ul li > section {
  display: inline-block;
  width: 50%;
  vertical-align: top;
}

.c0 > ul li > section:nth-child(odd) {
  padding-right: 15px;
}

.c0 > ul li > section:nth-child(even) {
  padding-left: 15px;
}

.c0 > ul li > section > p {
  font-size: 13px;
  color: #333740;
  font-weight: 500;
}

.c0 > ul li > section > div:first-of-type {
  height: 34px;
}

.c0 > ul li > section > div:first-of-type > div:first-of-type {
  height: 34px;
  min-height: 34px;
  border: 1px solid #e3e9f3;
  border-radius: 2px;
  font-size: 13px;
  color: #333740;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
}

.c0 > ul li > section > div:first-of-type > div:first-of-type > div:first-of-type {
  height: 32px;
  padding: 0 1rem;
}

.c0 > ul li > section > div:first-of-type > div:first-of-type > div:last-of-type {
  display: none;
}

.c0 > ul li > section > div:first-of-type > div:first-of-type:hover {
  cursor: text;
}

.c0 > ul li > section > div:first-of-type > span + div:first-of-type {
  border-color: #78caff;
  box-shadow: none;
}

.c0 > ul li > section + div {
  position: absolute;
  top: 7px;
  right: 0;
}

.c0 > ul li > section + div button {
  margin: 0;
}

.c0 .bordered input {
  border: 1px solid #f64d0a;
}

<div
    class="c0"
  >
    <ul>
      <li>
        <section>
          <p>
            <span>
              Key
            </span>
          </p>
        </section>
        <section>
          <p>
            <span>
              Value
            </span>
          </p>
        </section>
      </li>
      <li>
        <section>
          <div
            class=" css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1w9ejix-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1uccc91-singleValue"
                />
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="headers.0.key"
              type="hidden"
              value=""
            />
          </div>
        </section>
        <section>
          <div
            class="c1"
          >
            <input
              autocomplete="off"
              class="c2"
              id="headers.0.value"
              name="headers.0.value"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </section>
        <div>
          <button
            class="c3"
            type="button"
          />
        </div>
      </li>
    </ul>
    <button
      type="button"
    >
      <svg
        fill="#007eff"
        height="13px"
        viewBox="0 0 13 13"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <path
            d="M6.5.5v12m6-6H.5"
            fill="none"
            stroke="#007eff"
            stroke-linecap="round"
          />
        </g>
      </svg>
      <span>
        Create a new header
      </span>
    </button>
  </div>
</DocumentFragment>
`;
